Horses! 🐎🐎🐎

thechriskent opened the pull request:
Added the tiny horse pet with 11 colors for all your tiny horse needs!

|Tiny Horse|Color|
|----------|-----|
|![black_idle_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/e4fc5d3b-c207-4d26-856a-e663dd0bcff0)|Black|
|![brown_walk_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/656363ec-64cb-4bbc-8a63-9364ac0c8a8b)|Brown|
|![white_with_ball_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/d6b4a933-39b7-44d7-8c02-9354f4d48afa)|White|
|![paint_brown_idle_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/66e59b5e-dccf-44f8-ad51-26d38c146db5)|Paint Brown|
|![paint_black_swipe_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/5d46eb99-2efd-4224-8b1d-062a263206ac)|Paint Black|
|![paint_beige_walk_fast_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/4acab46e-8748-4810-8ba9-65ea9a9e061c)|Paint Beige|
|![socks_brown_walk_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/018f3cf9-2a4e-4c53-9555-c89c3a221a52)|Socks Brown|
|![socks_beige_idle_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/08c61188-3017-4290-b578-d840e4f4ed6a)|Socks Beige|
|![socks_black_run_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/1f28f3ad-752e-48c0-9d21-9f90770438c2)|Socks Black|
|![magical_idle_8fps](https://github.com/tonybaloney/vscode-pets/assets/8364109/470bb511-6d95-42b7-af06-a91c977bc4c9)|Magical|
|![warrior_swipe_8fps_Looped](https://github.com/tonybaloney/vscode-pets/assets/8364109/4f352e9e-e62f-4502-b049-ece64ff4c5f8)|Warrior|

Animations adapted from the beautiful [Horse Sprite With Rider Asset Pack](https://onfe.itch.io/horse-sprite-with-rider-asset-pack) by [Onfe](https://onfe.itch.io/) which is [free to use](https://onfe.itch.io/horse-sprite-with-rider-asset-pack/devlog/702600/horse-sprite-with-rider-asset-pack-is-now-free) with attribution.

I've been using this extension for years and kept thinking it would be cool if it had some horses - then I realized you accept PRs 😀. Please let me know any adjustments you'd like or if there's anything else you need from me to get the people the horses they demand. Thank you for the amazing and fun extension!

tonybaloney commented:
This is fantastic!!

Some feedback:

- Please can we have the pet type as "horse" instead of "tiny-horse". You can set the pet size to large, so they're pretty big (well not compared to an actual horse), but nothing else in the extension is tiny-xxx and they're all relatively the same size
- I'm not sure if "warrior" should be "red" as this is already a color.
- I'm not a fan of the climbing state and the jump down animation. I've never seen a horse climb vertically before. Please can we remove those.
- The horses are pretty active. The idle state is currently only designed for the left side, but it would be nice if they were idle on both sides. I grew up with horses and they spend most of the day eating grass. You could override the idle action to have longer idle times and make a flipped idle (see how the walk-right and walk-left actions work)

tonybaloney commented on src/panel/pets/tinyhorse.ts:
```diff
@@ -0,0 +1,236 @@
+    'Tony the Pony',
+    'Vigo Horsenberg',
+    'Ol Jetrho',
+    'Pony Soprano',
```
Thank you for not including Pie-oh-My

thechriskent commented:
Thank you for looking at this so quickly!

I've made several updates as requested:

- `tiny-horse` is now just `horse` everywhere!
  - Ironically, I only switched it to `tiny-horse` shortly before submitting the PR since they looked so small next to some of the other pets that they'd normally be way bigger than 😆. But makes sense to keep it consistent.
- Climbing related states are now gone.
-  Added stand left and right states for grazing. Also reworked the states to slow them down overall and emphasize grazing. Let me know if this isn't quite what you were looking for.

I'd like to keep the `warrior` "color" if possible as warrior horses is a very fun concept to me and one I use a lot for my open-source demos. But, this is your extension, and I will respect your opinion on the matter. Just let me know if you'd still like it updated to `red` and I'll take care of it right away.

tonybaloney commented:
I added another path into the run states as I noticed they weren't running unless you throw the ball. https://github.com/tonybaloney/vscode-pets/pull/572/commits/bab81018a77f1cc5077b80495fca8f17266419fd

In another change, I'd like to make the standing action a bit more dynamic and have each pet type be able to override it's "idleness". The cats are unrealistically active 🐱 

A fun side-effect of the friendship tree is that the horses will run around in herds sometimes. 

Other than that, LGTM! Thanks for the contribution

tonybaloney commented:
> I'd like to keep the warrior "color" if possible as warrior horses is a very fun concept to me

Sure. It's your PR 👍🏻

tonybaloney commented on l10n/bundle.l10n.json:
```diff
@@ -18,6 +19,18 @@
+	"orange": "Orange",
+	"blue": "Blue",
+	"akita": "Akita",
+	"socks black": "Black (Socks)",
```
Added the new colours so people can add the translations later
